// ngx-twitter-timeline/esm5/lib/ngx-twitter-timeline.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import * as i0 from "@angular/core";
var NgxTwitterTimelineService = /** @class */ (function () {
    function NgxTwitterTimelineService() {
        this.TWITTER_SCRIPT_ID = 'twitter-wjs';
        this.TWITTER_WIDGET_URL = 'https://platform.twitter.com/widgets.js';
    }
    /**
     * @return {?}
     */
    NgxTwitterTimelineService.prototype.loadScript = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return Observable.create((/**
         * @param {?} observer
         * @return {?}
         */
        function (observer) {
            _this.startScriptLoad();
            window['twttr'].ready((/**
             * @param {?} twttr
             * @return {?}
             */
            function (twttr) {
                observer.next(twttr);
                observer.complete();
            }));
        }));
    };
    /**
     * @private
     * @return {?}
     */
    NgxTwitterTimelineService.prototype.startScriptLoad = /**
     * @private
     * @return {?}
     */
    function () {
        window['twttr'] = ((/**
         * @param {?} d
         * @param {?} s
         * @param {?} id
         * @param {?} url
         * @return {?}
         */
        function (d, s, id, url) {
            /** @type {?} */
            var script;
            /** @type {?} */
            var firstScriptEl = d.getElementsByTagName(s)[0];
            /** @type {?} */
            var twitterScript = window['twttr'] || {};
            if (d.getElementById(id)) {
                return twitterScript;
            }
            script = d.createElement(s);
            script.id = id;
            script.src = url;
            firstScriptEl.parentNode.insertBefore(script, firstScriptEl);
            twitterScript._e = [];
            twitterScript.ready = (/**
             * @param {?} f
             * @return {?}
             */
            function (f) {
                twitterScript._e.push(f);
            });
            return twitterScript;
        })(document, 'script', this.TWITTER_SCRIPT_ID, this.TWITTER_WIDGET_URL));
    };
    NgxTwitterTimelineService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NgxTwitterTimelineService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgxTwitterTimelineService.ngInjectableDef = i0.defineInjectable({ factory: function NgxTwitterTimelineService_Factory() { return new NgxTwitterTimelineService(); }, token: NgxTwitterTimelineService, providedIn: "root" });
    return NgxTwitterTimelineService;
}());
export { NgxTwitterTimelineService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxTwitterTimelineService.prototype.TWITTER_SCRIPT_ID;
    /**
     * @type {?}
     * @private
     */
    NgxTwitterTimelineService.prototype.TWITTER_WIDGET_URL;
}
