// ngx-twitter-timeline/esm5/lib/ngx-twitter-timeline.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Input } from '@angular/core';
import { NgxTwitterTimelineService } from './ngx-twitter-timeline.service';
var NgxTwitterTimelineComponent = /** @class */ (function () {
    function NgxTwitterTimelineComponent(element, twitterTimelineService) {
        this.element = element;
        this.twitterTimelineService = twitterTimelineService;
        this.defaultOpts = {
            tweetLimit: 5
        };
        this.defaultData = {
            sourceType: 'url',
            url: 'https://twitter.com/twitterdev',
            screenName: 'twitterdev'
        };
    }
    ;
    /**
     * @return {?}
     */
    NgxTwitterTimelineComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        if (this.data && this.data.sourceType) {
            switch (this.data.sourceType) {
                case 'url':
                    delete this.defaultData.screenName;
                    break;
                case 'profile':
                    delete this.defaultData.url;
                    break;
                default:
                    break;
            }
            this.loadTwitterWidget();
        }
    };
    /**
     * @return {?}
     */
    NgxTwitterTimelineComponent.prototype.loadTwitterWidget = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.twitterTimelineService
            .loadScript()
            .subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        function (twttr) {
            /** @type {?} */
            var nativeElement = _this.element.nativeElement;
            nativeElement.innerHTML = "";
            window['twttr']
                .widgets
                .createTimeline(tslib_1.__assign({}, _this.defaultData, _this.data), nativeElement, tslib_1.__assign({}, _this.defaultOpts, _this.opts))
                .then((/**
             * @param {?} embed
             * @return {?}
             */
            function (embed) {
                // console.log(embed);
            }))
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            function (error) { return console.error(error); }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return console.error(err); }));
    };
    NgxTwitterTimelineComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-twitter-timeline',
                    template: ""
                }] }
    ];
    /** @nocollapse */
    NgxTwitterTimelineComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: NgxTwitterTimelineService }
    ]; };
    NgxTwitterTimelineComponent.propDecorators = {
        data: [{ type: Input }],
        opts: [{ type: Input }]
    };
    return NgxTwitterTimelineComponent;
}());
export { NgxTwitterTimelineComponent };
if (false) {
    /**
     * The data source definition object for the content to be displayed in the widget.
     * May be a widget ID string for a search API timeline or legacy widget
     * @type {?}
     */
    NgxTwitterTimelineComponent.prototype.data;
    /**
     * A hash of additional options to configure the widget
     * @type {?}
     */
    NgxTwitterTimelineComponent.prototype.opts;
    /** @type {?} */
    NgxTwitterTimelineComponent.prototype.defaultOpts;
    /** @type {?} */
    NgxTwitterTimelineComponent.prototype.defaultData;
    /**
     * @type {?}
     * @private
     */
    NgxTwitterTimelineComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    NgxTwitterTimelineComponent.prototype.twitterTimelineService;
    /* Skipping unhandled member: ;*/
}
