// ngx-twitter-timeline/fesm5/ngx-twitter-timeline.js
import { Observable } from 'rxjs';
import { __assign } from 'tslib';
import { Injectable, NgModule, defineInjectable, Component, ElementRef, Input } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxTwitterTimelineService = /** @class */ (function () {
    function NgxTwitterTimelineService() {
        this.TWITTER_SCRIPT_ID = 'twitter-wjs';
        this.TWITTER_WIDGET_URL = 'https://platform.twitter.com/widgets.js';
    }
    /**
     * @return {?}
     */
    NgxTwitterTimelineService.prototype.loadScript = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return Observable.create((/**
         * @param {?} observer
         * @return {?}
         */
        function (observer) {
            _this.startScriptLoad();
            window['twttr'].ready((/**
             * @param {?} twttr
             * @return {?}
             */
            function (twttr) {
                observer.next(twttr);
                observer.complete();
            }));
        }));
    };
    /**
     * @private
     * @return {?}
     */
    NgxTwitterTimelineService.prototype.startScriptLoad = /**
     * @private
     * @return {?}
     */
    function () {
        window['twttr'] = ((/**
         * @param {?} d
         * @param {?} s
         * @param {?} id
         * @param {?} url
         * @return {?}
         */
        function (d, s, id, url) {
            /** @type {?} */
            var script;
            /** @type {?} */
            var firstScriptEl = d.getElementsByTagName(s)[0];
            /** @type {?} */
            var twitterScript = window['twttr'] || {};
            if (d.getElementById(id)) {
                return twitterScript;
            }
            script = d.createElement(s);
            script.id = id;
            script.src = url;
            firstScriptEl.parentNode.insertBefore(script, firstScriptEl);
            twitterScript._e = [];
            twitterScript.ready = (/**
             * @param {?} f
             * @return {?}
             */
            function (f) {
                twitterScript._e.push(f);
            });
            return twitterScript;
        })(document, 'script', this.TWITTER_SCRIPT_ID, this.TWITTER_WIDGET_URL));
    };
    NgxTwitterTimelineService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NgxTwitterTimelineService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgxTwitterTimelineService.ngInjectableDef = defineInjectable({ factory: function NgxTwitterTimelineService_Factory() { return new NgxTwitterTimelineService(); }, token: NgxTwitterTimelineService, providedIn: "root" });
    return NgxTwitterTimelineService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxTwitterTimelineComponent = /** @class */ (function () {
    function NgxTwitterTimelineComponent(element, twitterTimelineService) {
        this.element = element;
        this.twitterTimelineService = twitterTimelineService;
        this.defaultOpts = {
            tweetLimit: 5
        };
        this.defaultData = {
            sourceType: 'url',
            url: 'https://twitter.com/twitterdev',
            screenName: 'twitterdev'
        };
    }
    /**
     * @return {?}
     */
    NgxTwitterTimelineComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        if (this.data && this.data.sourceType) {
            switch (this.data.sourceType) {
                case 'url':
                    delete this.defaultData.screenName;
                    break;
                case 'profile':
                    delete this.defaultData.url;
                    break;
                default:
                    break;
            }
            this.loadTwitterWidget();
        }
    };
    /**
     * @return {?}
     */
    NgxTwitterTimelineComponent.prototype.loadTwitterWidget = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.twitterTimelineService
            .loadScript()
            .subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        function (twttr) {
            /** @type {?} */
            var nativeElement = _this.element.nativeElement;
            nativeElement.innerHTML = "";
            window['twttr']
                .widgets
                .createTimeline(__assign({}, _this.defaultData, _this.data), nativeElement, __assign({}, _this.defaultOpts, _this.opts))
                .then((/**
             * @param {?} embed
             * @return {?}
             */
            function (embed) {
                // console.log(embed);
            }))
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            function (error) { return console.error(error); }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return console.error(err); }));
    };
    NgxTwitterTimelineComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-twitter-timeline',
                    template: ""
                }] }
    ];
    /** @nocollapse */
    NgxTwitterTimelineComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: NgxTwitterTimelineService }
    ]; };
    NgxTwitterTimelineComponent.propDecorators = {
        data: [{ type: Input }],
        opts: [{ type: Input }]
    };
    return NgxTwitterTimelineComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxTwitterTimelineModule = /** @class */ (function () {
    function NgxTwitterTimelineModule() {
    }
    NgxTwitterTimelineModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [NgxTwitterTimelineComponent],
                    imports: [],
                    exports: [NgxTwitterTimelineComponent]
                },] }
    ];
    return NgxTwitterTimelineModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgxTwitterTimelineService, NgxTwitterTimelineComponent, NgxTwitterTimelineModule };
